api/search.rb

```diff
@@ -1,58 +1,76 @@
 def get_thread_ids(context, group_ids, local_params, search_text)
-  filters = []
-  filters.push({term: {commentable_id: local_params['commentable_id']}}) if local_params['commentable_id']
-  filters.push({terms: {commentable_id: local_params['commentable_ids'].split(',')}}) if local_params['commentable_ids']
-  filters.push({term: {course_id: local_params['course_id']}}) if local_params['course_id']
+  must = []
+  filter = []
+  must.push({term: {commentable_id: local_params['commentable_id']}}) if local_params['commentable_id']
+  must.push({terms: {commentable_id: local_params['commentable_ids'].split(',')}}) if local_params['commentable_ids']
+  must.push({term: {course_id: local_params['course_id']}}) if local_params['course_id']
+  must.push(
+    {
+      multi_match: {
+        query: search_text,
+        fields: [:title, :body],
+        operator: :AND
+      }
+    }
+  )
+  group_id = local_params['group_id']
+
+  if group_id
+    filter.push(
+      {
+        bool: {
+          should: [
+            {:bool => {:must_not => {:exists => {:field => :group_id}}}},
+            {:term => {:group_id => group_id}}
+          ]
+        }
+      }
+    )
+  end
 
-  filters.push({or: [
-      {not: {exists: {field: :context}}},
-      {term: {context: context}}
-  ]})
+  filter.push(
+    {:bool => {:must_not => {:exists => {:field => context}}}},
+    {term: {context: context}}
+  )
 
   unless group_ids.empty?
-    filters.push(
-        {
-            bool: {
-                should: [
-                    {:not => {:exists => {:field => :group_id}}},
-                    {:terms => {:group_id => group_ids}}
-                ]
-            }
+    filter.push(
+      {
+        bool: {
+          should: [
+            {:bool => {:must_not => {:exists => {:field => :group_id}}}},
```

Comment:
Are we sure that we need nested boolean query with must_not option into should section? Root boolean query has must_not option too.  Maybe yes, just double checking.

Reply:
Done